roles/openshift_health_checker/openshift_checks/docker_image_availability.py

```diff
@@ -0,0 +1,76 @@
+# pylint: disable=missing-docstring
+from openshift_checks import OpenShiftCheck, get_var
+from openshift_checks.mixins import NotContainerizedMixin
+
+
+class DockerImageAvailability(NotContainerizedMixin, OpenShiftCheck):
+    """Check that required Docker images are available."""
+
+    name = "docker_image_availability"
+    tags = ["preflight"]
+
+    # attempt to pull required docker images and fail if
+    # any images are missing, or error occurs during pull
+    def run(self, tmp, task_vars):
+        openshift_release = get_var(task_vars, "openshift_release")
+        openshift_image_tag = get_var(task_vars, "openshift_image_tag")
+
+        changed = False
+        failed_pulls = set()
+
+        pulled_images = self.pull_images(self.add_image_tags(self.rpm_docker_images(), openshift_release), tmp, task_vars)
+        if "failed_images" in pulled_images:
+            failed_pulls.update(pulled_images["failed_images"])
+
+        if not openshift_image_tag:
+            if len(failed_pulls) < len(self.rpm_docker_images()) + len(self.rpm_qualified_docker_images()):
+                changed = True
+            return {"changed": changed, "failed": True, "msg": "Unable to retrieve fully qualified image tag. Failed to fetch images: %s" % (set(failed_pulls).update(self.rpm_qualified_docker_images()))}
+
+        # pull each required image that requires a qualified image
+        # tag (v0.0.0.0) rather than a standard release format tag (v0.0)
+        pulled_qualified_images = self.pull_images(self.add_image_tags(self.rpm_qualified_docker_images(), "v" + openshift_image_tag), tmp, task_vars)
```

Comment:
Would using skopeo make sense here to verify images without pulling them?

Output of `skopeo inspect docker://docker.io/openshift/origin`:
```
{
    "Name": "docker.io/openshift/origin",
    "Tag": "latest",
    "Digest": "sha256:7fe04a3bc9dc0508f846980888c948914663146c7cabc0dc0beb06169d75f21c",
    "RepoTags": [
        "b596f940a813a660be8982c37a2fdac29641b5f5",
        "beta1",
        "latest",
        "testing",
        "v0.2.1",
        "v0.2.2",
        "v0.3.1",
        "v0.3.2",
        "v0.3.3",
        "v0.3.4",
        "v0.3",
        "v0.4.1",
        "v0.4.2",
        "v0.4.3",
        "v0.4.4",
        "v0.4",
        "v0.5.1",
        "v0.5.2",
        "v0.5.3",
        "v0.5.4",
        "v0.5",
        "v0.6.1",
        "v0.6.2",
        "v0.6",
        "v1.0.0",
        "v1.0.1",
        "v1.0.2",
        "v1.0.3",
        "v1.0.4",
        "v1.0.5",
        "v1.0.6",
        "v1.0.7",
        "v1.0.8",
        "v1.1.1.1",
        "v1.1.1",
        "v1.1.2",
        "v1.1.3",
        "v1.1.4",
        "v1.1.5",
        "v1.1.6",
        "v1.1",
        "v1.2.0-rc1",
        "v1.2.0-rc2",
        "v1.2.0",
        "v1.2.1",
        "v1.2.2",
        "v1.3.0-alpha.0",
        "v1.3.0-alpha.1",
        "v1.3.0-alpha.2",
        "v1.3.0-alpha.3",
        "v1.3.0-rc1",
        "v1.3.0",
        "v1.3.1",
        "v1.3.2",
        "v1.3.3",
        "v1.4.0-alpha.0",
        "v1.4.0-alpha.1",
        "v1.4.0-rc1",
        "v1.4.0",
        "v1.4.1",
        "v1.5.0-alpha.0",
        "v1.5.0-alpha.1",
        "v1.5.0-alpha.2",
        "v1.5.0-alpha.3"
    ],
    "Created": "2017-02-28T01:26:25.908410645Z",
    "DockerVersion": "1.12.6",
    "Labels": {
        "build-date": "20161214",
        "io.k8s.description": "OpenShift Origin is a platform for developing, building, and deploying containerized applications. See https://docs.openshift.org/latest for more on running OpenShift Origin.",
        "io.k8s.display-name": "OpenShift Origin Application Platform",
        "license": "GPLv2",
        "name": "CentOS Base Image",
        "vendor": "CentOS"
    },
    "Architecture": "amd64",
    "Os": "linux",
    "Layers": [
        "sha256:a3ed95caeb02ffe68cdd9fd84406680ae93d633cb16422d00e8a7c22955b46d4",
        "sha256:45a2e645736c4c66ef34acce2407ded21f7a9b231199d3b92d6c9776df264729",
        "sha256:a3ed95caeb02ffe68cdd9fd84406680ae93d633cb16422d00e8a7c22955b46d4",
        "sha256:a3ed95caeb02ffe68cdd9fd84406680ae93d633cb16422d00e8a7c22955b46d4",
        "sha256:e24e870d9c831b7ee97c3f763d4118acea93781746b797665ceb4fb0fc327e25",
        "sha256:a3ed95caeb02ffe68cdd9fd84406680ae93d633cb16422d00e8a7c22955b46d4",
        "sha256:6a8f42f1ab1e81a4365cce09ea66bc65e30f436b2754ab879242c1fe9ace11b8",
        "sha256:190f15d3619652dbfdbf3a8d666f3b04bfeb27138050aa5fa993601af462c802",
        "sha256:8a54e0c0e9d1ebd81c6915f8fdac5a14a6b4c0a0786a4721b879c032d3e4efaa",
        "sha256:ebfb90719f35b6847ab381c48b887942c09621a5b4392e82d7f18601bf71922b",
        "sha256:a3ed95caeb02ffe68cdd9fd84406680ae93d633cb16422d00e8a7c22955b46d4",
        "sha256:a3ed95caeb02ffe68cdd9fd84406680ae93d633cb16422d00e8a7c22955b46d4",
        "sha256:a3ed95caeb02ffe68cdd9fd84406680ae93d633cb16422d00e8a7c22955b46d4",
        "sha256:a3ed95caeb02ffe68cdd9fd84406680ae93d633cb16422d00e8a7c22955b46d4",
        "sha256:a3ed95caeb02ffe68cdd9fd84406680ae93d633cb16422d00e8a7c22955b46d4"
    ]
}
```